rsconnect_jupyter/static/connect.js

```diff
@@ -1626,14 +1713,48 @@ define([
       // allow raw html
       sanitize: false,
 
-      open: function() {
-
+    open: function() {
+        // TODO: Use this in the conda support branch
+        var compatibilityMode = false;
+        var forceGenerate = false;
+        var hasRequirementsTxt = false;
+        function prepareRequirementsTxtDialog(hasRequirements) {
+          var info = dialog.find('#write-manifest-overwrite-info');
+          if (!hasRequirements) {
+            info.html(
+               '<p><i class="fa fa-question-circle"></i> A <span class="code">requirements.txt</span> file was not found in the notebook ' +
+              'directory. One will be generated automatically from your current python environment.</p>'
+            );
+          } else {
+            info.html(
+                '<div id="requirements-txt-container">' +
+                '<input type="radio" name="requirements-txt" id="use-existing" value="use-existing" checked />' +
+                '<label for="use-existing">' +
+                ' Use the existing <span class="code">requirements.txt</span> file in the notebook directory.' +
+                '</label><br />' +
+                '<input type="radio" name="requirements-txt" id="generate-new" value="generate-new" />' +
+                '<label for="generate-new">' +
+                '  Generate a <span class="code">requirements.txt</span> from the current python environment.' +
+                '</label></div>'
+            );
+          }
+        }
         dialog.find('#version-info').html(
             'rsconnect-jupyter server extension version: ' +
             rsconnectVersionInfo.rsconnect_jupyter_server_extension + '<br />' +
             'rsconnect-jupyter nbextension version: ' + rsconnectVersionInfo.js_version + '<br />' +
             'rsconnect-python version:' + rsconnectVersionInfo.rsconnect_python_version
         );
+        ContentsManager.list_contents(notebookDirectory)
+            .then(function (contents) {
+              for(var index in contents.content) {
+                if (contents.content[index].name === 'requirements.txt') {
+                  hasRequirementsTxt = true;
+                  break;
+                }
+              }
+              prepareRequirementsTxtDialog(hasRequirementsTxt);
+            });
```

Comment:
This seems to duplicate the code on lines 1112-1122 - should this be factored out?

Reply:
It's different, I'll rename them to make that clearer